tensorflow/cc/framework/gradients_test.cc

```diff
@@ -77,9 +106,10 @@ TEST_F(GradientsTest, OneMatMul) {
   for (const bool expected : {false, true}) {
     const Scope& scope = expected ? scope_expected_ : scope_test_;
     // Construct forward graph.
-    auto x = Const(scope, {{1.0, 2.0}, {3.0, 4.0}});
-    auto y = Const(scope, {{1.0, 0.0}, {0.0, 1.0}});
+    auto x = ThreeNodeVariable(scope, {{1.0, 2.0}, {3.0, 4.0}});
```

Comment:
@skye: I chose to edit the tests because as far as I know we don't differentiate w.r.t a Const. Else people will differentiate w.r.t a Const W1 for instance and then create an ApplyGradientDescent node that updates W1 using the collected gradient. Updating a Const is in opposition to the term const.

So a test passing with Const nodes is worthless because in reality it will not be done like that but with Variables (I maybe wrong).

Anyway, I'll revert the tests and add mine once a solution working with the case you described above is implemented.